src/main/java/com/slack/eithernet/ApiResult.kt

```diff
@@ -308,6 +308,15 @@ public object ApiResultCallAdapterFactory : CallAdapter.Factory() {
                   val withTag =
                     when (val result = response.body()) {
                       is Success -> result.withTags(result.tags + tags)
+                      null -> {
+                        val successType = apiResultType.actualTypeArguments[0]
+                        val responseCode = response.code()
+                        if ((responseCode == 204 || responseCode == 205) && successType == Unit::class.java) {
```

Comment:
Let's avoid the reflection until after we've checked the status code and add a comment for posterity

```suggestion
                        // No/reset content responses have no body
                        // So we can inline Unit responses
                        val responseCode = response.code()
                        if ((responseCode == 204 || responseCode == 205) && apiResultType.actualTypeArguments[0] == Unit::class.java) {
```

Reply:
Agree